src/main/scala/nl/knaw/dans/easy/fedora2vault/EasyFedora2vaultApp.scala

```diff
@@ -15,10 +15,71 @@
  */
 package nl.knaw.dans.easy.fedora2vault
 
+import java.io.InputStream
+
+import better.files.File
+import com.yourmediashelf.fedora.client.FedoraClient
+import javax.naming.ldap.InitialLdapContext
+import nl.knaw.dans.bag.v0.DansV0Bag
+import nl.knaw.dans.easy.fedora2vault.Command.FeedBackMessage
+import nl.knaw.dans.easy.fedora2vault.FoXml.{ getEmd, _ }
+
 import scala.util.{ Success, Try }
+import scala.xml.{ Elem, XML }
+
+class EasyFedora2vaultApp(configuration: Configuration) {
+  lazy val fedoraProvider: FedoraProvider = new FedoraProvider(new FedoraClient(configuration.fedoraCredentials))
+  lazy val ldapContext: InitialLdapContext = new InitialLdapContext(configuration.ldapEnv, null)
+  private lazy val ldap = new Ldap(ldapContext)
 
-class EasyFedora2vaultApp(configuration: Configuration)  {
+  def simpleTransform(datasetId: DatasetId, outputDir: File): Try[FeedBackMessage] = {
 
+    def managedMetadataStream(foXml: Elem, streamId: String, bag: DansV0Bag, metadataFile: String) = {
+      managedStreamLabel(foXml, streamId)
+        .map { label =>
+          val extension = label.split("[.]").last
+          val bagFile = s"$metadataFile.$extension"
+          fedoraProvider.disseminateDatastream(datasetId, streamId)
+            .map(bag.addMetadataStream(bagFile))
+            .tried.flatten
+        }
+    }
 
+    def compareManifest(bag: DansV0Bag)(streamId: String): Try[Any] = {
+      fedoraProvider.disseminateDatastream(datasetId, streamId)
+        .map { inputStream: InputStream =>
+          val manifestst = bag.payloadManifests
+          Success(()) // TODO EASY-2678
+        }.tried.flatten
+    }
 
+    for {
+      foXml <- fedoraProvider.getObject(datasetId).map(XML.load).tried
+      depositor <- getOwner(foXml)
+      bag <- DansV0Bag.empty(outputDir).map(_.withEasyUserAccount(depositor))
+      _ <- getEmd(foXml)
+        .flatMap(bag.addMetadataXml("emd.xml"))
+      _ <- getAmd(foXml)
+        .flatMap(bag.addMetadataXml("amd.xml"))
+      _ <- getDdm(foXml)
+        .map(bag.addMetadataXml("dataset.xml"))
+        .getOrElse(Success(())) // TODO EASY-2683
+      _ <- getMessageFromDepositor(foXml)
+        .map(bag.addMetadataXml("depositor-info/message-from-depositor.txt"))
+        .getOrElse(Success(())) // TODO EMD/other/remark?
+      _ <- getFilesXml(foXml)
+        .map(bag.addMetadataXml("files.xml"))
```

Comment:
For future improvement:
the datamanager may have added extra metadata to the file objects or may even have deleted, added or moved files. Hence the `files.xml` (if it exists) may be outdated. Same story as with the DDM/EMD above.